### DISCUSSION TITLE
I have special PDF files here did not succeed in opening

### DISCUSSION THREAD

**Discussion by a1-ops** [jsvine/pdfplumber#661] [Ask for help with specific PDFs]
Hello, author, first of all I would like to thank you, because my job requires me to deal with PDF from time to time, and it is your work that drives me to complete one project after another. But recently, a new project has caused me more trouble than ever.
This is a new PDF file, by double-clicking to open what is obviously a very common file, but can not be opened by your work, will produce the following error.

**struct.error: unpack requires a buffer of 6 bytes**

Here is the code I ran.

```python
pdf = pdfplumber.open(pdf_ads)
for page in range(0, len(pdf.pages)):
    page_info = pdf.pages[page]
    print(page_info.objects) 
```

And this error will occur on the last line。
I opened the PDF file can almost be counted by ten thousand, but this error is really the first time and I can not do anything about it. So I would like to ask the author to try to open this file, and if you can open it, you can give the corresponding code and environment.
thanks.

Originally I just wanted to upload the original report and end it, but given that there is a lot of sensitive personal information mixed in the report, I have tried to make reductions here. Then the weirdest thing happened, that is, the redacted PDF file can be successfully opened, I do not know what happened, so I intend to upload the original report and the redacted report, I hope the author can find the answer, but also because of the sensitivity of the report also hope that the author does not over-distribute, thanks again.

The following is the original document.
[pdf.pdf](https://github.com/jsvine/pdfplumber/files/8821723/pdf.pdf)
The following files are redacted
[fix_pdf.pdf](https://github.com/jsvine/pdfplumber/files/8821726/fix_pdf.pdf)

**Comment by jsvine** [comment]
Thanks for your kind words about this library, @a1-ops. Looking at the error traceback, the issue seems to caused by some invalid aspect of the PDF's fonts. Specifically, the PDF processing is failing on this line of `pdfminer.six`, this library's main dependency: https://github.com/pdfminer/pdfminer.six/blob/86e34873e4daeb7329aa7420208e82f8d41e38b6/pdfminer/cmapdb.py#L136

If you [repair the PDF with Ghostscript](https://superuser.com/a/282056), the error should disappear. Does that work for you?

**Reply by a1-ops** [reply]
Thank you for your reply, I successfully installed ghostscript in python, but I don't know how to use it, and there are few tutorials online, could you please share the code to repair PDF documents.

**Reply by jsvine** [reply]
Hi @a1-ops — the instructions for repair are in the link above: https://superuser.com/a/282056

Specifically, this part:

```
 gs \
  -o repaired.pdf \
  -sDEVICE=pdfwrite \
  -dPDFSETTINGS=/prepress \
   corrupted.pdf
```

Just replace `corrupted.pdf` with the path to the file you want to repair, and `repaired.pdf` with the path where you want the repaired PDF to be written.
